addons/account/wizard/account_payment_register.py

```diff
@@ -946,6 +948,7 @@ def _create_payments(self):
         payments = self._init_payments(to_process, edit_mode=edit_mode)
         self._post_payments(to_process, edit_mode=edit_mode)
         self._reconcile_payments(to_process, edit_mode=edit_mode)
+        self.group_payment = self._context.get('force_group_payment')
```

Comment:
I was gonna agree with you but actually when multiple aml are selected to register a payment, nothing tries to batch them so if you just select and register payment, you will get an expected singleton error because it was expecting a single line to register a payment for. This line makes sure the lines are batched so the creation of the payment can handle multiple lines.

Reply:
But isn't the compute function responsible to do what you describe ? This line in particular assigns the field after computation of the payments; I don't get why.